ui/src/core_components/embed/CoreMapbox.vue

```diff
@@ -0,0 +1,207 @@
+<template>
+	<div ref="rootEl" class="CoreMapbox">
+		<div ref="mapEl" class="map" />
+		<div class="mask" />
+	</div>
+</template>
+
+<script lang="ts">
+import { FieldType } from "../../streamsyncTypes";
+import { cssClasses } from "../../renderer/sharedStyleFields";
+import * as mapboxgl from "mapbox-gl";
+
+const markersDefaultData = [
+	{ lat: 37.79322359164316, lng: -122.39999318828129, name: "Marker" },
+];
+
+export default {
+	streamsync: {
+		name: "Mapbox",
+		description:
+			"A component to embed a Mapbox map. It can be used to display a map with markers.",
+		docs: "For this component you need Mapbox access token: https://www.mapbox.com/api-documentation/#access-tokens-and-token-scopes",
+		category: "Embed",
+		fields: {
+			accessToken: {
+				name: "Access Token",
+				default: "",
+				desc: "Access token from Mapbox",
+				type: FieldType.Text,
+			},
+			mapStyle: {
+				name: "Map style",
+				default: "mapbox://styles/mapbox/standard",
+				type: FieldType.Text,
+				desc: "Map style URL",
+			},
+			zoom: {
+				name: "Zoom",
+				default: "8",
+				type: FieldType.Number,
+			},
+			lat: {
+				name: "Latitude",
+				default: "37.79322359164316",
+				type: FieldType.Number,
+			},
+			lng: {
+				name: "Longitude",
+				default: "-122.39999318828129",
+				type: FieldType.Number,
+			},
+			markers: {
+				name: "Markers",
+				default: JSON.stringify(markersDefaultData, null, 2),
+				desc: "",
+				type: FieldType.Object,
+			},
+			cssClasses,
+		},
+		events: {
+			"mapbox-marker-click": {
+				desc: "Capture single clicks on markers.",
+			},
+			"mapbox-click": {
+				desc: "Capture single click on map.",
+			},
+		},
+	},
+};
+</script>
+
+<script setup lang="ts">
+import "mapbox-gl/dist/mapbox-gl.css";
+import { inject, ref, watch, computed } from "vue";
+import injectionKeys from "../../injectionKeys";
+const fields = inject(injectionKeys.evaluatedFields);
+const rootEl = ref(null);
+const mapEl = ref(null);
+const center = computed<mapboxgl.LngLatLike>(() => [
+	fields.lng.value,
+	fields.lat.value,
+]);
+let map = null;
+let markers: mapboxgl.Marker[] = [];
+
+const initMap = async () => {
+	if (!mapEl.value) return;
+	if (!fields.accessToken.value) return;
+	const mapboxgl = await import("mapbox-gl");
```

Comment:
I used only `npm run docs:dev` and it was working fine but now when i ran `docs:build` it failed because in quickstart-tutorial I had link to localhost.
To quickly solve it I added following line to vitepress config
```js
 ignoreDeadLinks: 'localhostLinks',
```

Reply:
I think we should add docs build to CI

Reply:
We use Netlify for docs, I think it can be easily hooked up to Github to build automatically. We're also on a Pro plan so the build allowance is quite generous.

I'm not sure if you had other idea in mind with CI for docs, keen to hear.